socorro/cron/jobs/monitoring.py

```diff
@@ -0,0 +1,171 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import json
+import os
+from collections import namedtuple
+from os.path import dirname
+from subprocess import PIPE, Popen
+
+from configman import Namespace
+from crontabber.base import BaseCronApp
+
+from socorro.lib import raven_client
+
+
+REPO_ROOT = dirname(dirname(dirname(dirname(__file__))))
+
+
+VulnerabilityBase = namedtuple('Vulnerability', (
+    'type',
+    'dependency',
+    'installed_version',
+    'affected_versions',
+    'description',
+))
+
+
+class Vulnerability(VulnerabilityBase):
+    @property
+    def key(self):
+        return '[%s] %s' % (self.type, self.dependency)
+
+    @property
+    def summary(self):
+        return 'Installed: %s; Affected: %s; %s' % (
+            self.installed_version,
+            self.affected_versions,
+            self.description,
+        )
```

Comment:
Periodically, I think about using `attrs` for things like this. Glyph wrote a blog post about it which I look back on periodically:

https://glyph.twistedmatrix.com/2016/08/attrs.html

If we find ourselves doing more of these, maybe we should look into switching to `attrs`.